### PR TITLE
Upgrade to Spring Boot 3.2.0 / Spring Data Elasticsearch 5.2.0

### PR DESCRIPTION
**Pull request by reta**
### Description
Upgrade to Spring Boot 3.2.0 / Spring Data Elasticsearch 5.2.0

### Issues Resolved
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by jose-hernandez-iop** [comment]
It would be great to see this merged in so that I can upgrade to SpringBoot 3.2.0

File: spring-data-opensearch/src/main/java/org/opensearch/data/client/orhlc/RestClients.java [hunk 1]
```diff
@@ -165,57 +145,6 @@ default void close() throws IOException {
         }
     }
 
-    /**
-     * Logging interceptors for OpenSearch client logging.
-     *
-     * @see ClientLogger
-     */
-    private static class HttpLoggingInterceptor implements HttpResponseInterceptor, HttpRequestInterceptor {
```
[the remark this answers is not in the document]
Comment (reta) [reply spring-data-opensearch/src/main/java/org/opensearch/data/client/orhlc/RestClients.java:173]: This functionality was removed in Spring Data Elasticsearch 5.2

File: spring-data-opensearch/src/main/java/org/opensearch/data/core/AbstractIndexTemplate.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+import org.springframework.util.Assert;
+
+/**
+ * Base implementation of {@link IndexOperations} common to Transport and Rest based Implementations of IndexOperations.
+ *
+ * @author Peter-Josef Meisch
+ * @author Sascha Woo
+ * @since 4.0
+ */
+public abstract class AbstractIndexTemplate implements IndexOperations {
```
[the remark this answers is not in the document]
Comment (reta) [reply spring-data-opensearch/src/main/java/org/opensearch/data/core/AbstractIndexTemplate.java:46]: This functionality was removed in Spring Data Elasticsearch 5.2, the class was (and is used) by OpenSearch and covered by tests (it just changes the codebase)

**Review by dlvenable — APPROVED** [review]

**Comment by dlvenable** [comment]
Thanks @reta for the explanations.

**Comment by rajadilipkolli** [comment]
Hi @reta , when will the spring boot starter gets released?

**Comment by reta** [comment]
> Hi @reta , when will the spring boot starter gets released?

Working on it as we speak :)

**Pull request merged**